aspects/aspects.go

```diff
@@ -112,61 +112,70 @@ func NewAspectDirectory(name string, aspects map[string]interface{}, dataBag Dat
 			return nil, errors.New("cannot create aspect without access patterns")
 		}
 
-		aspect := &Aspect{
-			Name:           name,
-			accessPatterns: make([]*accessPattern, 0, len(aspectPatterns)),
-			directory:      aspectDir,
+		aspect, err := newAspect(aspectDir, name, aspectPatterns)
+		if err != nil {
+			return nil, fmt.Errorf("cannot create aspect %q: %w", name, err)
```

Comment:
perhaps s/create/define/ ?

Reply:
I see we used "create" already in other errors, no very strong opinion, but define could be better across the board here